src/directives/formly-field.js

```diff
@@ -77,9 +77,41 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
         return undefined
       }
       if (angular.isDefined(newVal)) {
-        $scope.model[$scope.options.key] = newVal
+        parseSet($scope.options.key, $scope.model, newVal)
+      }
+      return parseGet($scope.options.key, $scope.model)
+    }
+
+    function parseSet(key, model, newVal) {
+      // If either of these are null/undefined then just return undefined
+      if (!key || !model) {
+        return
+      }
+      // If we are working with a number then $parse wont work, default back to the old way for now
+      if (angular.isNumber(key)) {
+        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+        model[key] = newVal
+      } else {
+        const setter = $parse($scope.options.key).assign
+        if (setter) {
```

Comment:
Kent, I tried the commands you sent but it didn’t work.

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Monday, November 2, 2015 at 9:15 AM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(): added  functionality (#531)


In src/directives/formly-field.js:
> +      }
> +      return parseGet($scope.options.key, $scope.model)
> +    }
> +
> +    function parseSet(key, model, newVal) {
> +      // If either of these are null/undefined then just return undefined
> +      if (!key || !model) {
> +        return
> +      }
> +      // If we are working with a number then $parse wont work, default back to the old way for now
> +      if (angular.isNumber(key)) {
> +        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
> +        model[key] = newVal
> +      } else {
> +        const setter = $parse($scope.options.key).assign
> +        if (setter) {
That's reasonable. Sounds good. If you could fix the multiple commits issue, that'd be great. Thanks
On Mon, Nov 2, 2015, 7:06 AM MCKRUZ <notifications@github.com> wrote: In src/directives/formly-field.js <#531 (comment)> :  > +      } > +      return parseGet($scope.options.key, $scope.model) > +    } > + > +    function parseSet(key, model, newVal) { > +      // If either of these are null/undefined then just return undefined > +      if (!key || !model) { > +        return > +      } > +      // If we are working with a number then $parse wont work, default back to the old way for now > +      if (angular.isNumber(key)) { > +        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric > +        model[key] = newVal > +      } else { > +        const setter = $parse($scope.options.key).assign > +        if (setter) {  I would actually leave it. Because if you get into a situation where the key you sent in is non numeric but doesn't satisfy a property constraint you will blow up if you try to use a setter where one isn't present. It wont "work" but it also wont fail either. Your call.  — Reply to this email directly or view it on GitHub <https://github.com/formly-js/angular-formly/pull/531/files#r43630610>. 
-- - Kent C. Dodds <https://twitter.com/kentcdodds>
—
Reply to this email directly or view it on GitHub.